lib/vdc/DCMPAS.cpp

```diff
@@ -353,6 +370,32 @@ int DCMPAS::initialize(
 	rc = _InitDataVars(ncdfc) ;
 	if (rc<0) return(-1);
 
+
+	// If u and v variables are present create derived Uzonal and Umeridional
+	// variables.
+	//
+	if (
+		ncdfc->VariableExists(uNormalVarName) && 
+		ncdfc->VariableExists(uTangentialVarName)) {
+
+		for (auto varname : {"Uzonal", "Umeridional"}) {
+			bool zonalFlag = (varname == string("Uzonal"));
+			DerivedZonalMeridonal *derivedVar = new DerivedZonalMeridonal(
+				varname, this, ncdfc, uNormalVarName, uTangentialVarName,
+				zonalFlag
+			);
+
+			rc = derivedVar->Initialize();
+			if (rc<0) return(-1);
+
+			_dvm.AddDataVar(derivedVar);
+
+			DC::DataVar dvarInfo;
+			(void) derivedVar->GetDataVarInfo(dvarInfo);
+			_dataVarsMap[varname] = dvarInfo;
+		}
+	}
+
```

Comment:
Also, when this function finishes in a normal state, who's responsible for freeing the memory allocated by new?

Reply:
The DCMPAS destructor frees all of the dynamically created derived variables.